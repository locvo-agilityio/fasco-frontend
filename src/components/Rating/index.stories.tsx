import type { Meta, StoryObj } from '@storybook/react';

// Components
import Rating from '.';

const meta = {
  title: 'Components/Rating',
  component: Rating,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Rating>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    rating: 5,
  },
};

export const Secondary: Story = {
  args: {
    rating: 3,
    color: 'black',
  },
};
