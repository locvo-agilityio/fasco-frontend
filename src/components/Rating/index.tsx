import { Star } from 'lucide-react';

// Utils
import { cn } from '@/utils';

interface IRatingProps {
  rating?: number;
  size?: number;
  color?: 'yellow' | 'black';
}

const Rating = ({ rating = 5, size = 19, color = 'yellow' }: IRatingProps) => {
  const isYellow = color === 'yellow';
  const fillColor = isYellow
    ? 'fill-yellow-500 text-yellow-500'
    : 'fill-black text-black';
  const strokeColor = isYellow
    ? 'stroke-yellow-500 text-white'
    : 'stroke-black text-white';

  return (
    <div className="flex gap-1">
      {[...Array(5)].map((_, index) => (
        <Star
          size={size}
          key={index}
          className={cn(index < rating ? fillColor : strokeColor)}
        />
      ))}
    </div>
  );
};

export default Rating;
