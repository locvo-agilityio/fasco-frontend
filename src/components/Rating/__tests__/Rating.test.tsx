import { render } from '@testing-library/react';

// Components
import Rating from '..';

describe('Rating', () => {
  it('should render correctly', () => {
    const { container } = render(<Rating color="black" />);

    expect(container).toMatchSnapshot();
  });
});
