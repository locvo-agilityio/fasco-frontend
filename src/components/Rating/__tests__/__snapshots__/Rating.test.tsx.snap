// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render correctly 1`] = `
<div>
  <div
    class="flex gap-1"
  >
    <svg
      class="lucide lucide-star fill-black text-black"
      fill="none"
      height="19"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="19"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
    <svg
      class="lucide lucide-star fill-black text-black"
      fill="none"
      height="19"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="19"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
    <svg
      class="lucide lucide-star fill-black text-black"
      fill="none"
      height="19"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="19"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
    <svg
      class="lucide lucide-star fill-black text-black"
      fill="none"
      height="19"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="19"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
    <svg
      class="lucide lucide-star fill-black text-black"
      fill="none"
      height="19"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="19"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
  </div>
</div>
`;
